### ISSUE TITLE
Create Serialization to/from Dictionary

### ISSUE THREAD

**Issue by drumnkyle** [drumnkyle/music-notation-swift#75]
Create a `toDictionary()` and `init(dictionary: [String: AnyObject])` for every main type. These 2 methods should live in an extension in the same file as the type is defined.

We are doing this to allow for serialization to/from JSON so that music notation can be shared and stored.

**Comment by drumnkyle** [comment]
We have decided that doing it as a dictionary is probably not of great use, because this will not be a long-lived format. Instead we are exploring either making a binary file or a human readable one that would be easy for people to create from scratch with a text editor.

For the time being, we are thinking binary would be good, because it will make file sizes small, which will be useful for transfer of data over slower networks. @migue48 and I discussed possibly using Protobuf, which would potentially give us the benefit of binary without having to implement from scratch and also JSON style debugging.

We are exploring if this will go into a different project or what considering we are trying to keep this library at 0 dependencies.

**Comment by btc** [comment]
What about exporting to MusicXML format?

https://wpmedia.musicxml.com/wp-content/uploads/2012/12/musicxml-tutorial.pdf

**Comment by drumnkyle** [comment]
Yes, we plan to implement support for import and export of MusicXML as well. However, it is a somewhat complicated format, so it is a little lower priority at this point. It is very important though, because it is an open format and one that will allow anything made with this library to work with other applications.

I thought we had an issue already, but I will create one now to track that.
